sysvars/sysvars_ec2.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Didi Kohen
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package sysvars
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func getEC2Meta(sysVars map[string]string) error {
+	ec2Session, err := session.NewSession()
+	if err != nil {
+		return fmt.Errorf("Could not create session %v", err)
+	}
+	ec2met := ec2metadata.New(ec2Session)
+	// Doing the availability check in module since we need a session
+	if ec2met.Available() == false {
+		sysVars["EC2_METADATA_Availabile"] = "false"
+		return nil
+	}
+
+	id, err := ec2met.GetInstanceIdentityDocument()
+	if err != nil {
+		sysVars["EC2_METADATA_Availabile"] = "false"
+		return fmt.Errorf("Could not get instance document %v", err)
+	}
+
+	sysVars["EC2_METADATA_Availabile"] = "true"
```

Comment:
Instead of adding all the variables by default, I'd start with a few that are useful for probing (also, some of this information may be security risk, e.g. DevpayProductCodes, AccountID).

Can you start with only: AvailabilityZone, Region, InstanceID, PrivateIP and may be ImageID?

Reply:
Removed sensitive info, added ramdisk and architecture to the list